swarm/src/lib.rs

```diff
@@ -77,28 +77,33 @@ pub use protocols_handler::{
     SubstreamProtocol
 };
 
+
+pub type BoxSubstream = Substream<StreamMuxerBox>;
```

Comment:
This name is a bit misleading I think. The `StreamMuxer` is boxed, but the `Substream` is not. Considering that this type alias is not used often, maybe we can do without it?

Reply:
It's a bit confusing. The `Substream<T>` alias is defined as `muxing::SubstreamRef<Arc<TMuxer>>`, and `SubstreamRef` holds a `Arc<TMuxer>` and one of the substreams.

Essentially, `BoxSubstream` contains an `Arc` and a `usize` representing the substream.

I'll remove both `BoxSubstream` and `Substream`.